### PR TITLE
Windows - MinGW - gcc 8.1

### PR DESCRIPTION
**Pull request by qPCR4vir**
Hi, 
this was what I needed to run all SeqAn3 test in windows.

### CODE REVIEW

**Comment by qPCR4vir** [comment]
related to https://github.com/simongog/sdsl-lite/issues/212 ?

File: include/sdsl/int_vector_mapper.hpp [hunk 1]
```diff
@@ -346,7 +346,7 @@ class temp_file_buffer {
 	static std::string tmp_file(const std::string& dir)
 	{
 		char tmp_file_name[1024] = {0};
-#ifdef MSVC_COMPILER
+#ifdef  _WIN32  // MSVC_COMPILER
```
Comment (eseiler) [inline include/sdsl/int_vector_mapper.hpp]:
```suggestion
#ifdef _WIN32
```

File: include/sdsl/memory_management.hpp [hunk 2]
```diff
@@ -291,7 +291,7 @@ inline void block_markfree(mm_block_t* ptr) { block_update(ptr, SETFREE(ptr->siz
 /* mark the block as used */
 inline void block_markused(mm_block_t* ptr) { block_update(ptr, UNMASK_SIZE(ptr->size)); }
 
-#ifndef MSVC_COMPILER
+#ifndef _WIN32   //  MSVC_COMPILER
```
Comment (eseiler) [inline include/sdsl/memory_management.hpp]:
```suggestion
#ifndef _WIN32
```

File: include/sdsl/memory_management.hpp [hunk 3]
```diff
@@ -713,7 +713,7 @@ class memory_manager {
 public:
 	static uint64_t* alloc_mem(size_t size_in_bytes)
 	{
-#ifndef MSVC_COMPILER
+#ifndef _WIN32   // MSVC_COMPILER
```
Comment (eseiler) [inline include/sdsl/memory_management.hpp]:
```suggestion
#ifndef _WIN32
```

File: include/sdsl/memory_management.hpp [hunk 4]
```diff
@@ -723,7 +723,7 @@ class memory_manager {
 	}
 	static void free_mem(uint64_t* ptr)
 	{
-#ifndef MSVC_COMPILER
+#ifndef _WIN32   // MSVC_COMPILER
```
Comment (eseiler) [inline include/sdsl/memory_management.hpp]:
```suggestion
#ifndef _WIN32
```

File: include/sdsl/memory_management.hpp [hunk 5]
```diff
@@ -734,7 +734,7 @@ class memory_manager {
 	}
 	static uint64_t* realloc_mem(uint64_t* ptr, size_t size)
 	{
-#ifndef MSVC_COMPILER
+#ifndef _WIN32   //  MSVC_COMPILER
```
Comment (eseiler) [inline include/sdsl/memory_management.hpp]:
```suggestion
#ifndef _WIN32
```

File: include/sdsl/memory_management.hpp [hunk 6]
```diff
@@ -746,12 +746,14 @@ class memory_manager {
 public:
 	static void use_hugepages(size_t bytes = 0)
 	{
-#ifndef MSVC_COMPILER
+#ifndef _WIN32   // MSVC_COMPILER
```
Comment (eseiler) [inline include/sdsl/memory_management.hpp]:
```suggestion
#ifndef _WIN32
```

File: include/sdsl/memory_management.hpp [hunk 7]
```diff
@@ -746,12 +746,14 @@ class memory_manager {
...
 		auto& m = the_manager();
 		hugepage_allocator::the_allocator().init(bytes);
 		m.hugepages = true;
 #else
-		throw std::runtime_error("hugepages not support on MSVC_COMPILER");
+		// avoid error: unused parameter 'bytes' [-Werror=unused-parameter]
+	    throw std::runtime_error(std::string("hugepages not support on _WIN32- not set bytes: ") +
+	                             std::to_string(bytes) );
```
Comment (eseiler) [inline include/sdsl/memory_management.hpp]:
```suggestion
	    throw std::runtime_error(std::string("hugepages not supported on Windows"));
	    (void) bytes;
```
I'd just change the error message to say "Windows" and `(void)` the `bytes`. There's not really a point in printing `bytes`.

File: include/sdsl/memory_management.hpp [hunk 8]
```diff
@@ -826,7 +828,7 @@ class memory_manager {
 			return file_content.data();
 		}
 		memory_monitor::record(file_size);
-#ifdef MSVC_COMPILER
+#ifdef _WIN32  // MSVC_COMPILER
```
Comment (eseiler) [inline include/sdsl/memory_management.hpp]:
```suggestion
#ifdef _WIN32
```

File: include/sdsl/memory_management.hpp [hunk 9]
```diff
@@ -871,7 +873,7 @@ class memory_manager {
 			return 0;
 		}
 		memory_monitor::record(-((int64_t)size));
-#ifdef MSVC_COMPILER
+#ifdef  _WIN32  // MSVC_COMPILER
```
Comment (eseiler) [inline include/sdsl/memory_management.hpp]:
```suggestion
#ifdef _WIN32
```

File: include/sdsl/memory_management.hpp [hunk 10]
```diff
@@ -898,7 +900,7 @@ class memory_manager {
 		if (is_ram_file(fd)) {
 			return ram_fs::truncate(fd, new_size);
 		}
-#ifdef MSVC_COMPILER
+#ifdef  _WIN32  // MSVC_COMPILER
```
Comment (eseiler) [inline include/sdsl/memory_management.hpp]:
```suggestion
#ifdef _WIN32
```

File: include/sdsl/memory_tracking.hpp [hunk 11]
```diff
@@ -29,14 +29,16 @@
...
+#            define NOMINMAX 1
+#     endif
+# include <windows.h>
+# include <io.h>
 #else
-#include <sys/mman.h>
-#include <unistd.h> // for getpid, file_size, clock_gettime
+# include <sys/mman.h>
+# include <unistd.h> // for getpid, file_size, clock_gettime
 #endif
```
Comment (eseiler) [inline include/sdsl/memory_tracking.hpp]: Can you remove the indentation?

File: include/sdsl/memory_tracking.hpp [hunk 12]
```diff
@@ -29,14 +29,16 @@
 #include <sstream>
 #include <fstream>
 
-#ifdef MSVC_COMPILER
-// windows.h has min/max macro which causes problems when using std::min/max
-#define NOMINMAX
-#include <windows.h>
-#include <io.h>
+#ifdef _WIN32
+#     ifndef  NOMINMAX
```
Comment (eseiler) [inline include/sdsl/memory_tracking.hpp]:
```suggestion
#ifndef NOMINMAX
```

File: include/sdsl/util.hpp [hunk 13]
```diff
@@ -45,12 +45,15 @@
 #include <typeinfo>
 
 
-#ifndef MSVC_COMPILER
-#include <sys/time.h>	 // for struct timeval
+#ifndef  MSVC_COMPILER
```
Comment (eseiler) [inline include/sdsl/util.hpp]:
```suggestion
#ifndef MSVC_COMPILER
```

File: include/sdsl/util.hpp [hunk 14]
```diff
@@ -45,12 +45,15 @@
...
+#   include <cxxabi.h>
```
Comment (eseiler) [inline include/sdsl/util.hpp]:
```suggestion
#include <cxxabi.h>
```

File: include/sdsl/util.hpp [hunk 15]
```diff
@@ -197,10 +200,16 @@ inline size_t file_size(const std::string& file)
 inline std::string basename(std::string file)
 {
 	file = disk_file_name(file); // remove RAM-prefix
-#ifdef MSVC_COMPILER
+#ifdef _WIN32  // MSVC_COMPILER
```
Comment (eseiler) [inline include/sdsl/util.hpp]:
```suggestion
#ifdef _WIN32
```

File: include/sdsl/util.hpp [hunk 16]
```diff
@@ -197,10 +200,16 @@ inline size_t file_size(const std::string& file)
...
 	char* c						= _strdup((const char*)file.c_str());
 	char  file_name[_MAX_FNAME] = {0};
-	::_splitpath_s(c, NULL, 0, NULL, NULL, file_name, _MAX_FNAME, NULL, 0);
+#   ifdef 		MSVC_COMPILER
+	    ::_splitpath_s(c, NULL, 0, NULL, NULL, file_name, _MAX_FNAME, NULL, 0);
+#   else
+	    ::_splitpath  (c, NULL,    NULL,       file_name,             NULL   );
+ //_splitpath(*_FullPath,*_Drive, *_Dir,      *_Filename,            *_Ext    ) __MINGW_ATTRIB_DEPRECATED_SEC_WARN;
+#   endif
+
```
Comment (eseiler) [inline include/sdsl/util.hpp]:
```suggestion
#ifdef MSVC_COMPILER
	::_splitpath_s(c, NULL, 0, NULL, NULL, file_name, _MAX_FNAME, NULL, 0);
#else
	::_splitpath(c, NULL, NULL, file_name, NULL);
#endif
```

File: include/sdsl/util.hpp [hunk 17]
```diff
@@ -218,11 +227,17 @@ inline std::string dirname(std::string file)
 {
 	bool ram_file = is_ram_file(file);
 	file		  = disk_file_name(file); // remove RAM-prefix
-#ifdef MSVC_COMPILER
+
+#ifdef _WIN32  // MSVC_COMPILER
```
Comment (eseiler) [inline include/sdsl/util.hpp]:
```suggestion
#ifdef _WIN32
```

File: include/sdsl/util.hpp [hunk 18]
```diff
@@ -218,11 +227,17 @@ inline std::string dirname(std::string file)
...
 	char* c					 = _strdup((const char*)file.c_str());
 	char  dir_name[_MAX_DIR] = {0};
 	char  drive[_MAX_DRIVE]  = {0};
-	::_splitpath_s(c, drive, _MAX_DRIVE, dir_name, _MAX_DIR, NULL, 0, NULL, 0);
+#   ifdef 		MSVC_COMPILER
+    	::_splitpath_s(c, drive, _MAX_DRIVE, dir_name, _MAX_DIR, NULL, 0, NULL, 0);
+#   else
+    	::_splitpath  (c, drive,             dir_name,           NULL,    NULL   );
+    //_splitpath(*_FullPath,*_Drive,        *_Dir,      *_Filename,     *_Ext    ) __MINGW_ATTRIB_DEPRECATED_SEC_WARN;
+#   endif
```
Comment (eseiler) [inline include/sdsl/util.hpp]:
```suggestion
#ifdef MSVC_COMPILER
	::_splitpath_s(c, NULL, 0, NULL, NULL, file_name, _MAX_FNAME, NULL, 0);
#else
	::_splitpath(c, NULL, NULL, file_name, NULL);
#endif
```

**Comment by qPCR4vir** [comment]
?

**Pull request merged**